src/frontend/src/pages/PIAListPage/index.tsx

```diff
@@ -30,27 +34,65 @@ const PIAList = () => {
   const [searchText, setSearchText] = useState(
     searchParams.get('searchText') || '',
   );
+  const setSearchParamsForSearchText = () => {
+    const params: any = {};
+
+    params.searchText = searchText;
+    if (searchParams.get('filterPiaDrafterByCurrentUser'))
+      params.filterPiaDrafterByCurrentUser = filterPiaDrafterByCurrentUser;
+    if (searchParams.get('filterByStatus'))
+      params.filterByStatus = filterByStatus;
+    if (searchParams.get('filterByMinistry'))
+      params.filterByMinistry = filterByMinistry;
+
+    for (const [key, value] of searchParams.entries()) {
+      params[key] = value;
+    }
```

Comment:
aren't line 41-46, and 48-49 doing same things? 🤔

Reply:
yes, i am working on this issue. trying to find a way can call function outside of useEffect hook